scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5158 @@
+                            the logarithm
+            - ``'quadrature'``: numerically log-integrate the logarithm of the
+                                integrand
+
+            Not all `method` options are available for all distributions.
+            If the selected `method` is not available, a ``NotImplementedError``
+            will be raised.
+
+        Returns
+        -------
+        out : array
+            The logarithm of the differential entropy of the random variable.
+
+        See Also
+        --------
+        entropy
+        logpdf
+
+        Notes
+        -----
+        If the entropy of a distribution is negative, then the logarithm of
+        entropy is complex with imaginary part divisible by :math:`\pi`. For
+        consistency, the result of this function always has complex dtype,
+        regardless of the value of the imaginary part.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-1., b=1.)
+
+        Evaluate the logarithm of the differential entropy:
+
+        >>> X.logentropy()
+        (-0.3665129205816642+0j)
+        >>> np.allclose(np.exp(X.logentropy()), X.entropy())
+        True
+
+        For a random variable with negative entropy, the logarithm of the
+        entropy has an imaginary part equal to `np.pi`.
+
+        >>> X = stats.Uniform(a=-.1, b=.1)
+        >>> X.entropy(), X.logentropy()
+        (-1.6094379124341007, (0.4758849953271105+3.141592653589793j))
+
+        """
+        return self._logentropy_dispatch(method=method, **self._parameters) + 0j
+
+    @_dispatch
+    def _logentropy_dispatch(self, method=None, **kwargs):
+        if self._overrides('_logentropy_formula'):
+            method = self._logentropy_formula
+        elif self.tol is _null and self._overrides('_entropy_formula'):
+            method = self._logentropy_logexp
+        else:
+            method = self._logentropy_quadrature
+        return method
+
+    def _logentropy_formula(self, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _logentropy_logexp(self, **kwargs):
+        res = np.log(self._entropy_dispatch(**kwargs)+0j)
+        return _log_real_standardize(res)
+
+    def _logentropy_quadrature(self, **kwargs):
+        def logintegrand(x, **kwargs):
+            logpdf = self._logpdf_dispatch(x, **kwargs)
+            return logpdf + np.log(0j+logpdf)
+        res = self._quadrature(logintegrand, kwargs=kwargs, log=True)
+        return _log_real_standardize(res + np.pi*1j)
+
+    @_set_invalid_nan_property
+    def entropy(self, *, method=None):
+        r"""Differential entropy
+
+        In terms of probability density function :math:`f(x)` and its support
+        :math:`\chi`, the differential entropy of a random variable :math:`X` is:
```

Comment:
Maybe throw in a reference, https://en.wikipedia.org/wiki/Differential_entropy?

Unfortunately there's no pithy way to describe differential entropy. It doesn't actually have nice properties, and is just kind of the naive continuous analog of Shannon entropy.

Reply:
Yeah, I've read about that vs LDDF. And I've wondered how the numerical value is used for a given distribution. Does someone need to know the numerical value of the differential entropy of a relativistic Breit-Wigner distribution, for example?